### PR TITLE
Added event capturing option to DOMEmitterMixin.

### LINKED ISSUE #76: Lack of useCapture option in DOMEmitterMixin
There is no way to set `useCapture` option in `DOMEmitterMixin`
https://github.com/ckeditor/ckeditor5-ui/blob/master/src/domemittermixin.js#L70

I need this option in `FocusManager` for listening `focus` and `blur` events which don't bubble. Unfortunately `focusin` and `focusout` (the bubbling ones) are not supported by FF.

### PR DESCRIPTION
**Pull request by oskarwrobel**
Fixes #76

### CODE REVIEW

File: src/domemittermixin.js [hunk 1]
```diff
@@ -52,10 +52,12 @@ extend( ProxyEmitter.prototype, EmitterMixin, {
...
 	 * @param {Object} [options.context] The object that represents `this` in the callback. Defaults to the object firing the event.
+	 * @param {Boolean} [options.useCapture=false] Indicates that events of this type will be dispatched to the registered
+	 * listener before being dispatched to any EventTarget beneath it in the DOM tree.
 	 *
 	 * @method ui.ProxyEmitter#on
 	 */
-	on( event ) {
+	on( event, callback, options = {} ) {
 		// Execute parent class method first.
 		EmitterMixin.on.apply( this, arguments );
```
Comment (Reinmar) [inline src/domemittermixin.js]: Don't use `arguments` if you don't have to.

Comment (Reinmar) [inline src/domemittermixin.js]: By commenting on this now I mean that we can easily fix it. You could also do the same with `listenTo` in fact.

**Comment by oskarwrobel** [comment]
There is a lack of `on()` test at all. Hm... I think we we can test `on()` instead of `listenTo()` and then test if `listenTo()` uses `on()` with proper emitter.

**Comment by oskarwrobel** [comment]
There is also no tests for `off()` method. I'll handle it.

**Comment by oskarwrobel** [comment]
Ok I understand why there is no tests for `on()` and `off()` methods. They can't be directly called in DOM context. Only `listenTo()` and `stopListening()` can call them in DOM context. Direct call of `on()` and `off()` if fully tested in `EmitterMixin` tests.

**Comment by Reinmar** [comment]
I don't understand. Can't we check that a listener added using `on()` is executed when we fire a native event (I saw we can do this)?

**Comment by oskarwrobel** [comment]
To test `element.on()` we need to mix this mixin into some native element but I'm afraid it won't work.

**Comment by Reinmar** [comment]
Yeah, that's a bad idea.

**Pull request merged**